test failure seems unrelated
```
2022-03-17 08:08:18,090 400 ERROR odoo odoo.addons.account_invoice_import_invoice2data.tests.test_invoice_import: ERROR: TestInvoiceImport.test_import_free_invoice
Traceback (most recent call last):
  File "/__w/edi/edi/account_invoice_import_invoice2data/wizard/account_invoice_import.py", line 53, in invoice2data_parse_invoice
    invoice2data_res = extract_data(file_name, templates=templates)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/main.py", line 93, in extract_data
    return t.extract(optimized_str)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/invoice_template.py", line 204, in extract
    result = parsers.regex.parse(self, {"regex": v, "type": "date"}, optimized_str, True)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/parsers/regex.py", line 36, in parse
    result[k] = template.coerce_type(v, settings["type"])
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/invoice_template.py", line 151, in coerce_type
    return self.parse_date(value)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/invoice_template.py", line 136, in parse_date
    languages=self.options["languages"],
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/conf.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/__init__.py", line 61, in parse
    data = parser.get_date_data(date_string, date_formats)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 429, in get_date_data
    locale, date_string, date_formats, settings=self._settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 178, in parse
    return instance._parse()
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 182, in _parse
    date_data = self._parsers[parser_name]()
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 196, in _try_freshness_parser
    return freshness_date_parser.get_date_data(self._get_translated_date(), self._settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 235, in _get_translated_date
    self.date_string, keep_formatting=False, settings=self._settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/languages/locale.py", line 131, in translate
    relative_translations = self._get_relative_translations(settings=settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/languages/locale.py", line 158, in _get_relative_translations
    self._generate_relative_translations(normalize=True))
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/languages/locale.py", line 172, in _generate_relative_translations
    pattern = DIGIT_GROUP_PATTERN.sub(r'?P<n>\d+', pattern)
  File "/opt/odoo-venv/lib/python3.6/site-packages/regex/regex.py", line 700, in _compile_replacement_helper
    is_group, items = _compile_replacement(source, pattern, is_unicode)
  File "/opt/odoo-venv/lib/python3.6/site-packages/regex/_regex_core.py", line 1736, in _compile_replacement
    raise error("bad escape \\%s" % ch, source.string, source.pos)
regex._regex_core.error: bad escape \d at position 7
```

--- a/edi_oca/models/edi_exchange_record.py
+++ b/edi_oca/models/edi_exchange_record.py
@@ -17,6 +17,7 @@ class EDIExchangeRecord(models.Model):
     _inherit = "mail.thread"
     _description = "EDI exchange Record"
     _order = "exchanged_on desc"
+    _rec_name = "identifier"
 
     name = fields.Char(compute="_compute_name")
     identifier = fields.Char(required=True, index=True, readonly=True)

--- a/edi_oca/models/edi_exchange_record.py
+++ b/edi_oca/models/edi_exchange_record.py
@@ -112,9 +112,8 @@ class EDIExchangeRecord(models.Model):
     @api.depends("type_id.code", "model", "res_id")
     def _compute_name(self):
         for rec in self:
-            rec.name = "{} - {}".format(
-                rec.type_id.name, rec.record.name if rec.model else "Unrelated"
-            )
+            bits = [rec.type_id.name, rec.record.name if rec.model else ""]
+            rec.name = " - ".join([x for x in bits if x and x.strip()])
 
     @api.depends("model", "type_id")
     def _compute_exchange_filename(self):

--- a/edi_oca/views/edi_exchange_record_views.xml
+++ b/edi_oca/views/edi_exchange_record_views.xml
@@ -4,13 +4,13 @@
         <field name="model">edi.exchange.record</field>
         <field name="arch" type="xml">
             <tree string="EDI Exchange Record">
-                <field name="name" />
-                <field name="identifier" />
-                <field name="external_identifier" />
-                <field name="type_id" />
                 <field name="backend_id" />
-                <field name="res_id" groups="base.group_no_one" />
-                <field name="model" groups="base.group_no_one" />
+                <field name="identifier" />
+                <field name="external_identifier" optional="hide" />
+                <field name="name" />
+                <field name="type_id" />
+                <field name="res_id" groups="base.group_no_one" optional="hide" />
+                <field name="model" groups="base.group_no_one" optional="hide" />
                 <field name="exchanged_on" />
                 <field name="ack_received_on" />
                 <field
@@ -22,6 +22,7 @@
             </tree>
         </field>
     </record>
+
     <record id="edi_exchange_record_view_form" model="ir.ui.view">
         <field name="model">edi.exchange.record</field>
         <field name="arch" type="xml">
@@ -98,7 +99,7 @@
                     <group name="wrapper">
                         <group name="config">
                             <field name="type_id" />
-                            <field name="backend_id" />
+                            <field name="backend_id" options="{'no_create': True}" />
                         </group>
                         <group name="status" string="Status">
                             <field name="exchanged_on" />
@@ -164,10 +165,64 @@
             </form>
         </field>
     </record>
+
     <record id="edi_exchange_record_view_search" model="ir.ui.view">
         <field name="model">edi.exchange.record</field>
         <field name="arch" type="xml">
             <search string="EDI Exchange Record">
+                <filter
+                    string="Successful"
+                    name="filter_success"
+                    domain="[('edi_exchange_state','in', ['output_sent_and_processed', 'input_processed'])]"
+                />
+                <filter
+                    string="Failed"
+                    name="filter_failed"
+                    domain="[('edi_exchange_state','in', ['validate_error', 'output_error_on_send', 'output_sent_and_error', 'input_receive_error', 'input_processed_error'])]"
+                />
+                <filter
+                    string="Pending"
+                    name="filter_pending"
+                    domain="[('edi_exchange_state','in', ['output_pending', 'input_pending', 'output_sent', 'input_received'])]"
+                />
+                <separator />
+                <filter
+                    string="Inbound"
+                    name="filter_inbound"
+                    domain="[('type_id.direction','=', 'input')]"
+                />
+                <filter
+                    string="Outbound"
+                    name="filter_outbound"
+                    domain="[('type_id.direction','=', 'output')]"
+                />
+                <separator />
+                <filter
+                    string="Created today"
+                    name="filter_created_today"
+                    domain="[('create_date','&gt;=',  context_today().strftime('%%Y-%%m-%%d 00:00:00'))]"
+                    help="Show all records created today"
+                />
+                <filter
+                    string="Created last 7 days"
+                    name="filter_created_last_week"
+                    domain="[('create_date','&gt;=',  (context_today() - datetime.timedelta(days=7)).strftime('%%Y-%%m-%%d'))]"
+                    help="Show all records created in the last 7 days"
+                />
+                <filter
+                    string="Exchanged today"
+                    name="filter_exchanged_today"
+                    domain="[('exchanged_on','&gt;=', context_today().strftime('%%Y-%%m-%%d 00:00:00'))]"
+                    help="Show all records created today"
+                />
+                <filter
+                    string="Exchanged last 7 days"
+                    name="filter_exchanged_last_week"
+                    domain="[('exchanged_on','&gt;=', (context_today() - datetime.timedelta(days=7)).strftime('%%Y-%%m-%%d'))]"
+                    help="Show all records created in the last 7 days"
+                />
+                <separator />
+                <field name="backend_id" />
                 <field name="type_id" />
                 <field name="res_id" />
                 <field name="exchanged_on" />
@@ -178,6 +233,11 @@
                 <field name="ack_exchange_id" />
                 <field name="parent_id" />
                 <group expand="0" string="Group By">
+                    <filter
+                        name="group_by_backend_id"
+                        string="Backend"
+                        context="{'group_by': 'backend_id'}"
+                    />
                     <filter
                         name="group_by_type_id"
                         string="Type"
@@ -193,19 +253,85 @@
                         string="Parent exchange"
                         context="{'group_by': 'parent_id'}"
                     />
+                    <filter
+                        name="group_by_create_date"
+                        string="Create date"
+                        context="{'group_by': 'create_date'}"
+                    />
+                    <filter
+                        name="group_by_exchanged_on"
+                        string="Exchange date"
+                        context="{'group_by': 'exchanged_on'}"
+                    />
+                    <filter
+                        name="group_by_model"
+                        string="Model"
+                        context="{'group_by': 'model'}"
+                    />
                 </group>
             </search>
         </field>
     </record>
+
     <record model="ir.actions.act_window" id="act_open_edi_exchange_record_view">
-        <field name="name">EDI Exchange Record</field>
+        <field name="name">Exchanges</field>
         <field name="type">ir.actions.act_window</field>
         <field name="res_model">edi.exchange.record</field>
         <field name="view_mode">tree,form</field>
         <field name="search_view_id" ref="edi_exchange_record_view_search" />
         <field name="domain">[]</field>
-        <field name="context">{}</field>
+        <field name="context">{'search_default_filter_created_today': 1}</field>
     </record>
+
+    <record
+        model="ir.actions.act_window"
+        id="act_open_edi_exchange_record_view_pending"
+    >
+        <field name="name">Exchanges - pending</field>
+        <field name="type">ir.actions.act_window</field>
+        <field name="res_model">edi.exchange.record</field>
+        <field name="view_mode">tree,form</field>
+        <field name="search_view_id" ref="edi_exchange_record_view_search" />
+        <field name="domain">[]</field>
+        <field name="context">{'search_default_filter_pending': 1}</field>
+    </record>
+
+    <record model="ir.actions.act_window" id="act_open_edi_exchange_record_view_failed">
+        <field name="name">Exchanges - failed</field>
+        <field name="type">ir.actions.act_window</field>
+        <field name="res_model">edi.exchange.record</field>
+        <field name="view_mode">tree,form</field>
+        <field name="search_view_id" ref="edi_exchange_record_view_search" />
+        <field name="domain">[]</field>
+        <field name="context">{'search_default_filter_failed': 1}</field>
+    </record>
+
+    <record
+        model="ir.actions.act_window"
+        id="act_open_edi_exchange_record_view_inbound"
+    >
+        <field name="name">Exchanges - inbound</field>
+        <field name="type">ir.actions.act_window</field>
+        <field name="res_model">edi.exchange.record</field>
+        <field name="view_mode">tree,form</field>
+        <field name="search_view_id" ref="edi_exchange_record_view_search" />
+        <field name="domain">[]</field>
+        <field name="context">{'search_default_filter_inbound': 1}</field>
+    </record>
+
+    <record
+        model="ir.actions.act_window"
+        id="act_open_edi_exchange_record_view_outbound"
+    >
+        <field name="name">Exchanges - outbound</field>
+        <field name="type">ir.actions.act_window</field>
+        <field name="res_model">edi.exchange.record</field>
+        <field name="view_mode">tree,form</field>
+        <field name="search_view_id" ref="edi_exchange_record_view_search" />
+        <field name="domain">[]</field>
+        <field name="context">{'search_default_filter_outbound': 1}</field>
+    </record>
+
     <record
         model="ir.actions.act_window.view"
         id="act_open_edi_exchange_record_view_form"
@@ -215,6 +341,7 @@
         <field name="view_mode">form</field>
         <field name="view_id" ref="edi_exchange_record_view_form" />
     </record>
+
     <record
         model="ir.actions.act_window.view"
         id="act_open_edi_exchange_record_view_tree"

--- a/edi_oca/views/menuitems.xml
+++ b/edi_oca/views/menuitems.xml
@@ -1,18 +1,45 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <odoo>
     <menuitem
-        id="menu_edi_backend"
-        parent="base_edi.menu_edi_root"
-        name="Backends"
-        sequence="10"
-        action="act_open_edi_backend_view"
-    />
-    <menuitem
-        id="menu_edi_exchange_record"
+        id="menu_edi_exchange_record_root"
         parent="base_edi.menu_edi_root"
         name="Exchanges"
         sequence="100"
+    />
+    <menuitem
+        id="menu_edi_exchange_record_all"
+        parent="menu_edi_exchange_record_root"
+        name="All"
+        sequence="100"
         action="act_open_edi_exchange_record_view"
+    />
+    <menuitem
+        id="menu_edi_exchange_record_pending"
+        parent="menu_edi_exchange_record_root"
+        name="Failed"
+        sequence="200"
+        action="act_open_edi_exchange_record_view_failed"
+    />
+    <menuitem
+        id="menu_edi_exchange_record_failed"
+        parent="menu_edi_exchange_record_root"
+        name="Pending"
+        sequence="300"
+        action="act_open_edi_exchange_record_view_pending"
+    />
+    <menuitem
+        id="menu_edi_exchange_record_inbound"
+        parent="menu_edi_exchange_record_root"
+        name="Inbound"
+        sequence="400"
+        action="act_open_edi_exchange_record_view_inbound"
+    />
+    <menuitem
+        id="menu_edi_exchange_record_outbound"
+        parent="menu_edi_exchange_record_root"
+        name="Outbound"
+        sequence="500"
+        action="act_open_edi_exchange_record_view_outbound"
     />
     <menuitem
         id="menu_edi_config"

